pkg/volume/nfs/nfs_test.go

```diff
@@ -118,6 +119,18 @@ func doTestPlugin(t *testing.T, spec *volume.Spec) {
 	if mounter == nil {
 		t.Errorf("Got a nil Mounter")
 	}
+	if nfsm, ok := mounter.(*nfsMounter); ok {
```

Comment:
@jsafrane  I avoided changing anything in the `pkg` itself for testing, so i came up with this way, I hope it suffice

Reply:
@aojea  do you have a comment on this `mock` ? .. I tried to avoid adding a `method` just to assert `un-exported` field

Reply:
well, to be honest, the test has exactly the same code that the functionality, the only way to fail is if one of those diverge in the future :)

just throwing an idea, what if the fix goes here
https://github.com/kubernetes/kubernetes/blob/4f6c595240b2c2277482c8c10a3d1ad4a177b73e/pkg/volume/nfs/nfs.go#L258 

it seems a better place that current one , if IPv6 you build the source with brackets for the server.
This way, you can mock `nfsMounter.mounter.MountSensitiveWithoutSystemd(` 
https://github.com/kubernetes/kubernetes/blob/4f6c595240b2c2277482c8c10a3d1ad4a177b73e/pkg/volume/nfs/nfs.go#L191

and assert the server ip in `mountOptions`

https://github.com/kubernetes/kubernetes/blob/4f6c595240b2c2277482c8c10a3d1ad4a177b73e/pkg/volume/nfs/nfs.go#L264

Reply:
@aojea  this way, the fix will affect only the `mounter` when it is being `setUp` not `created`. 

The problem arose in the first place, when `pods` consuming `nfs` could not find the resource, and 
I searched the usage of both `setUp` which affects the mounter only when used. On the other hand, `newMounter` is used by `kubelet` so when the resource is actually allocated. 

Please correct me if I am wrong 🙏🏽